### PR TITLE
Pptp 1338 group radio

### PR DESCRIPTION
**Pull request by RyanKellyHMRC**
* I have removed the LiabilityLinkHelper as I found it to be confusing 
* No routing code is duplicated but some controllers are dependent on others 
* Link parameters have been added to a number of pages
* Refactoring to remove duplicate code
* Refactoring to remove unnecessary use of Futures
* Some routing bugs fixed

#### Check list 
 - [x] `./precheck` was executed (Integration/Component/Unit tests)
 - [x] `sbt scalafmt test:scalafmt` was executed
 - [x] User Acceptance Tests (UAT) were run locally and have passed.
 - [x] No Accessibility errors/warnings reported by Wave

image

### CODE REVIEW

File: app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/CheckLiabilityDetailsAnswersController.scala [hunk 1]
```diff
@@ -32,18 +32,23 @@ class CheckLiabilityDetailsAnswersController @Inject() (
   authenticate: AuthAction,
   journeyAction: JourneyAction,
   mcc: MessagesControllerComponents,
-  page: check_liability_details_answers_page,
-  liabilityLinkHelper: LiabilityLinkHelper
+  startRegistrationController: StartRegistrationController,
+  page: check_liability_details_answers_page
 ) extends FrontendController(mcc) with I18nSupport {
```
Comment (SteveSugden) [inline app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/CheckLiabilityDetailsAnswersController.scala:37]: Could this be a `LiabilityController`?  .....

Comment (RyanKellyHMRC) [reply app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/CheckLiabilityDetailsAnswersController.scala:37]: Most certainly. Good spot.

**Review by SteveSugden — APPROVED** [review]

File: test/builders/RegistrationBuilder.scala [hunk 2]
```diff
@@ -151,4 +150,7 @@ trait RegistrationBuilder {
         )
       )
 
+  def withRegistrationType(registrationType: Option[RegType]): RegistrationModifier =
```
[the remark this answers is not in the document]
Comment (SteveSugden) [reply test/builders/RegistrationBuilder.scala:153]: I've realised you need to be able to pass in None.  So maybe the non-optional build method can be removed at some point.

File: app/uk/gov/hmrc/plasticpackagingtax/registration/views/check_liability_details_answers_page.scala.html [hunk 3]
```diff
@@ -108,10 +114,31 @@
...
+                            )
+                        ))
                     )
                 ).filterNot(_.value.content == Empty)
             )
         )
         @saveAndContinue()
     }
-}
+}@import uk.gov.hmrc.plasticpackagingtax.registration.forms.RegType.SINGLE_ENTITY
```
[the remark this answers is not in the document]
Comment (RyanKellyHMRC) [reply app/uk/gov/hmrc/plasticpackagingtax/registration/views/check_liability_details_answers_page.scala.html:144]: Done

File: test/builders/RegistrationBuilder.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (RyanKellyHMRC) [reply test/builders/RegistrationBuilder.scala:153]: Have removed the non-optional variant and fixed clients.

**Pull request merged**